### PR TITLE
Add tech filtering!

### PR DESCRIPTION
**Pull request by koddsson**
I wanted to filter by react.js projects! Came up short for projects though! :<

### CODE REVIEW

File: app/static/app.js [hunk 1]
```diff
@@ -22,7 +22,11 @@ angular.module('contribute', [
      * leave it on the server and pull it in with a full AJAX URL.
      */
     $routeProvider
-    .when('/examples', {
+    .when('/examples/', {
+        templateUrl: 'examples.html',
+        controller: 'ExamplesController'
+    })
+    .when('/examples/:tech', {
```
Comment (koddsson) [inline app/static/app.js]: Is there a way for me to add a optional argument in `ngRoute` without adding a new endpoint?

Comment (peterbe) [inline app/static/app.js]: `ngRoute` is really basic. I don't know if you can add a `?` maybe at the end.

**Comment by peterbe** [comment]
Hey! Great to see your face again. 

This looks like a really good start but what's missing is the UI to select the tech(s). (other than just adding "/react" to the end of the URL).
One thing that bothers me about that in general (and it's not your PR's fault) is that a user might be unsure that to call things. Is it "react" or "reactjs". Is it "go" or "golang". If someone "types in" that she's interested in "golang" but nothing appears (even though there are "go" tags) that would be bad.

**Comment by peterbe** [comment]
cc @Osmose might haven an opinion since he wants to do more with the data but not necessary here on the site about it.

**Comment by koddsson** [comment]
>Hey! Great to see your face again.

Likewise! I've missed doing OSS work! :)

>This looks like a really good start but what's missing is the UI to select the tech(s). (other than just adding "/react" to the end of the URL).

Ah, I'll see if I can't add some links to the page for that tonight.

> One thing that bothers me about that in general (and it's not your PR's fault) is that a user might be unsure that to call things. Is it "react" or "reactjs". Is it "go" or "golang". If someone "types in" that she's interested in "golang" but nothing appears (even though there are "go" tags) that would be bad.

Hmm.. maybe I can add a lookup object in there so for example the `react` "technology" would match both the search terms `react` and `react.js`.

**Comment by koddsson** [comment]
I added the searchbox on the examples page!
contrib

I didn't solve the react/react.js and go/golang problem though. With the current projects it doesn't seem to be a issue but as projects get added it might.

**Comment by peterbe** [comment]
Mr PR against your PR https://github.com/koddsson/contribute.json/pull/1

**Comment by peterbe** [comment]
THANK YOU!

Now if only there were some React projects in that list. :)

**Pull request merged**